clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5894,7 +5894,22 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
       bool isFp32SqrtFunction =
           (FuncName == "sqrt" && !getLangOpts().OffloadFP32PrecSqrt &&
            IsFloat32Type);
-      if (hasFPAccuracyFuncMap || hasFPAccuracyVal || isFp32SqrtFunction) {
+      bool ArgsTypeIsFloat = true;
+      // In sycl mode, functions' arguments of type half are expanded
+      // to pointer types. Exclude these functions from bein emitted
+      // as fpbuiltins.
+      if (!getLangOpts().OffloadFP32PrecSqrt ||
+          !getLangOpts().OffloadFP32PrecDiv) {
+        for (auto Arg : IRCallArgs) {
```

Comment:
nit (if applicable): to avoid some static analyzers complains about 'auto causes copy'.
```suggestion
        for (auto &Arg : IRCallArgs) {
```